File: PgAce.App/Services/ConfigService.cs
using System;
using System.IO;
using System.Security.Cryptography;
using System.Text;
using Newtonsoft.Json;
using PgAce.App.Models;

namespace PgAce.App.Services
{
    public class ConfigService
    {
        private readonly string _path = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "config.json");

        public AppConfig Load()
        {
            if (!File.Exists(_path))
            {
                return new AppConfig();
            }

            var json = File.ReadAllText(_path);
            var config = JsonConvert.DeserializeObject<AppConfig>(json);
            return config ?? new AppConfig();
        }

        public void Save(AppConfig config)
        {
            var json = JsonConvert.SerializeObject(config, Formatting.Indented);
            File.WriteAllText(_path, json);
        }

        public string Protect(string plain)
        {
            if (string.IsNullOrEmpty(plain)) return string.Empty;
            var data = Encoding.UTF8.GetBytes(plain);
            var encrypted = ProtectedData.Protect(data, null, DataProtectionScope.CurrentUser);
            return "dpapi:" + Convert.ToBase64String(encrypted);
        }

        public string Unprotect(string cipher)
        {
            if (string.IsNullOrEmpty(cipher) || !cipher.StartsWith("dpapi:")) return string.Empty;
            var data = Convert.FromBase64String(cipher.Substring(6));
            var decrypted = ProtectedData.Unprotect(data, null, DataProtectionScope.CurrentUser);
            return Encoding.UTF8.GetString(decrypted);
        }
    }
}

File: PgAce.App/Services/ConfigService.vb
Imports System.IO
Imports Newtonsoft.Json
Imports PgAce.App.Models

Namespace PgAce.App.Services
    Public Class ConfigService
        Private ReadOnly _path As String

        Public Sub New(path As String)
            _path = path
        End Sub

        Public Function Load() As ConfigRoot
            If Not File.Exists(_path) Then
                Return New ConfigRoot()
            End If

            Dim json = File.ReadAllText(_path)
            Dim cfg = JsonConvert.DeserializeObject(Of ConfigRoot)(json)
            If cfg Is Nothing Then
                Return New ConfigRoot()
            End If
            For Each profile In cfg.Profiles
                profile.Password = EncryptionService.Unprotect(profile.Password)
            Next
            Return cfg
        End Function

        Public Sub Save(config As ConfigRoot)
            Dim clone = JsonConvert.DeserializeObject(Of ConfigRoot)(JsonConvert.SerializeObject(config))

            If clone.Settings.SavePassword Then
                For Each profile In clone.Profiles
                    profile.Password = EncryptionService.Protect(profile.Password)
                Next
            Else
                For Each profile In clone.Profiles
                    profile.Password = String.Empty
                Next
            End If

            Dim json = JsonConvert.SerializeObject(clone, Formatting.Indented)
            File.WriteAllText(_path, json)
        End Sub
    End Class
End Namespace
